### PR TITLE
Fix brown paperbag indexing

### PR DESCRIPTION
**Pull request by ktf**
Fix brown paperbag indexing

### CODE REVIEW

**Comment by ktf** [comment]
@davidrohr @shahor02 I suspect this is what actually breaks the ROOT 6.30.01 tests.
 
@chiarazampolli Given `MatchITSTPCQC::finalize` was reading randomly in memory, maybe it would be a good idea to cherry pick this for async?

File: Detectors/GlobalTracking/src/MatchITSTPCQC.cxx [hunk 1]
```diff
@@ -811,48 +811,48 @@ void MatchITSTPCQC::finalize()
   // first we use denominators and nominators to set the TEfficiency; later they are scaled
 
   // some checks
-  for (int i = 0; i < matchType::SIZE; ++i) {
-    for (int i = 0; i < mPtDen[i]->GetNbinsX(); ++i) {
+  for (int ti = 0; ti < matchType::SIZE; ++ti) {
+    for (int i = 0; i < mPtDen[ti]->GetNbinsX(); ++i) {
       if (mPtDen[i]->GetBinContent(i + 1) < mPtNum[i]->GetBinContent(i + 1)) {
```
Comment (shahor02) [inline Detectors/GlobalTracking/src/MatchITSTPCQC.cxx]: Should be 
```
if (mPtDen[ti]->GetBinContent(i + 1) < mPtNum[ti]->GetBinContent(i + 1)) {
```
?

**Review by shahor02 — CHANGES_REQUESTED** [review]

File: Detectors/GlobalTracking/src/MatchITSTPCQC.cxx (hunk 1 above)
Comment (ktf) [reply Detectors/GlobalTracking/src/MatchITSTPCQC.cxx]: yes, of course. Fixed.

**Comment by ktf** [comment]
Currently obviously broken. Merging.

**Comment by chiarazampolli** [comment]
Thanks @ktf , the logs don't really matter, what matters is the "setEfficiency" but everything was fine so far, so I think that QC till now was showing the correct results (which does not mean that one should not fix it). When the "types" were introduced, I had also run a test to compare that the behaviour did not change compared to the previous one. I am anyway tagging @mfaggin , for his information.

**Comment by ktf** [comment]
@shahor02 I suspect there is more to this. It now crashes in:

```
[48133:qc-task-GLO-MTCITSTPC]:     /sw/slc8_x86-64/ROOT/v6-28-08-alice1-2/lib/libHist.so.6.28: ROOT::Math::WrappedFunction<Beta_interval_length const&>::DoEval(double) const
[48133:qc-task-GLO-MTCITSTPC]:     /sw/slc8_x86-64/O2/12458-slc8_x86-64-local1/lib/libO2GlobalTracking.so: o2::globaltracking::MatchITSTPCQC::finalize()
```

**Comment by ktf** [comment]
In principle the old code can write garbage anywhere in memory (as it does with the new ROOT). What worries me is that now that the logging code actually works as intended, there is a new crash (see the stacktrace I reported above), which is not in the printout code.

**Comment by chiarazampolli** [comment]
Which line is causing the crash? I don't see it in the log.

**Comment by ktf** [comment]
Probably inlined. It comes from hist/hist/src/TEfficiency.cxx, where there is a minimization being done in BetaShortestInterval which is used by a number of other methods.

**Comment by ktf** [comment]
Nevermind, probably a fluke due to a too early rebuild. The other PR seems fine now, so I would say this is all we need.

**Pull request merged**